### PR TITLE
feat: add property sync functionality for group components

### PR DESCRIPTION
**Pull request by kyubisation**
This PR allows `<sbb-action-group>` and `<sbb-link-list>` to define properties, which will be applied to the nested elements.
With this it is no longer necessary to apply certain properties to all nested elements as the parent element will take care of it.

`<sbb-action-group>`:

- `variant="block"` is applied to any nested `<sbb-link>`
- `button-size` is applied to nested `<sbb-button>` `size` property
- `link-size` is applied to nested `<sbb-link>` `size` property

`<sbb-link-list>`:

- `variant="block"` is applied to any nested `<sbb-link>`
- `size` (previously `textSize`) and `negative` is copied to nested `<sbb-link>`

BREAKING CHANGE: This changes the behavior of `<sbb-action-group>` and `<sbb-link-list>`, as they now overwrite properties of nested `<sbb-link>` (`variant`, `size` (renamed from `text-size`)) and `<sbb-button>` (`size`).

### CODE REVIEW

File: src/components/sbb-action-group/sbb-action-group.stories.js [hunk 1]
```diff
@@ -1,85 +1,77 @@
 import { h } from 'jsx-dom';
 import readme from './readme.md';
 
-const firstButtonTemplate = (size, alignSelf) => (
-  <sbb-button size={size} align-self={alignSelf} variant="secondary">
+const firstButtonTemplate = (alignSelf) => (
```
Comment (jeripeierSBB) [inline src/components/sbb-action-group/sbb-action-group.stories.js:4]: maybe it's better to call it secondaryButtonTemplate and the other just ButtonTemplate

File: src/components/sbb-dialog/readme.md [hunk 2]
```diff
@@ -30,50 +36,50 @@ In order to show a modal you need to call the `open(event?: PointerEvent)` metho
...
 // Title provided via property
 <sbb-dialog id="my-dialog" title-content="Title" title-back-button="true">
   Dialog content.
-  <div slot="action-group">
-    <sbb-action-group>
-      <sbb-button sbb-dialog-close>Abort</sbb-button>
-      <sbb-button>Confirm</sbb-button>
-    </sbb-action-group>
-  </div>
+  <sbb-action-group slot="action-group">
```
Comment (jeripeierSBB) [inline src/components/sbb-dialog/readme.md:62]: Nice catch :)

File: src/components/sbb-dialog/sbb-dialog.tsx [hunk 3]
```diff
@@ -331,6 +331,14 @@ export class SbbDialog implements ComponentInterface, AccessibilityProperties {
     );
   }
 
+  private _setActionGroupSize(): void {
+    this._element
+      .querySelectorAll(
+        'sbb-action-group[slot="action-group"],[slot="action-group"] > sbb-action-group'
+      )
+      .forEach((element: HTMLSbbActionGroupElement) => (element.size = 'm'));
```
Comment (jeripeierSBB) [inline src/components/sbb-dialog/sbb-dialog.tsx:339]: Are you sure that this is always true?

File: src/components/sbb-dialog/sbb-dialog.tsx [hunk 4]
```diff
@@ -331,6 +331,14 @@ export class SbbDialog implements ComponentInterface, AccessibilityProperties {
     );
   }
 
+  private _setActionGroupSize(): void {
```
Comment (jeripeierSBB) [inline src/components/sbb-dialog/sbb-dialog.tsx:334]: Should we add a test?

File: src/components/sbb-link-list/sbb-link-list.e2e.ts [hunk 5]
```diff
@@ -1,13 +1,53 @@
...
+
+    it('should update attributes with text-size m', async () => {
+      element.setAttribute('text-size', 'm');
+      await page.waitForChanges();
+      const links = await page.findAll('sbb-link-list sbb-link');
+      expect(links.every((l) => l.getAttribute('text-size') === 'm')).toBeTruthy();
+    });
+
+    it('should update attributes with negative', async () => {
+      element.setAttribute('negative', '');
```
Comment (jeripeierSBB) [inline src/components/sbb-link-list/sbb-link-list.e2e.ts:47]: maybe it's clearer to set it to "true" ?

Comment (kyubisation) [reply src/components/sbb-link-list/sbb-link-list.e2e.ts:47]: No, I don't think so, as this should work like this.

File: src/components/sbb-action-group/sbb-action-group.stories.js [hunk 6]
```diff
@@ -1,85 +1,75 @@
...
+  <sbb-action-group {...args}>{TemplateThreeElements(null, null, 'center')}</sbb-action-group>
 );
 
-const TemplateVerticalAllocation003FullWidth = ({ size, ...args }) => (
-  <sbb-action-group {...args}>{TemplateThreeElements(size, null, null, 'end')}</sbb-action-group>
+const TemplateVerticalAllocation003FullWidth = ({ ...args }) => (
+  <sbb-action-group {...args}>{TemplateThreeElements(null, null, 'end')}</sbb-action-group>
 );
 
 const buttonSize = {
```
Comment (jeripeierSBB) [inline src/components/sbb-action-group/sbb-action-group.stories.js:68]:
```suggestion
const size = {
```

**Review by jeripeierSBB — APPROVED** [review]
LGTM, one renaming suggestion

**Pull request merged**